src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,151 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(null, value, null);
+            tail = head;
+            size++;
+            return;
+        }
+
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail.next != null) {
+            tail.prev = tail;
+        }
+        tail.next = newNode;
```

Comment:
think about how to simplify this logic. Can you rewrite so you don't need to use return; in first if and write size++ twice in this method?